### DISCUSSION TITLE
Get Started

### DISCUSSION THREAD

**Discussion by ypapouin** [sbidoul/runboat#57] [Deployment]
I would like to create a self-hosted runboat to have a CI/CD environment for Odoo (with GitLab).

I read the `README.md` and the [wiki](https://github.com/sbidoul/runboat/wiki).
I don't have any experience on Kubernetes... and its seems to be my weakest link.
For the record, I first created a new Debian 11 VM, installed docker on it and k8s but it was looks like overkill.
After some reading, I discovered rancher with its webui so, I re-installed this VM with Ubuntu 20.04 server, installed docker and I ran rancher (with its self-hosted k3s) in a container with privileges.
But once again, I was not able to make the link with runboat.

Could you make a simple diagram to explain how everything is inter-connected ?

In the wiki, you talk about `microk8s` that is another kubernetes implementation by ubuntu, so I suppose that it could be the base for a detailed How-To. 

It is not in my habits to claim help of this type, but could you record all console steps, starting from a fresh Ubuntu 20.04 server install, to run a self-hosted runboat with a single node kubernetes cluster ?

**Comment by sbidoul** [comment]
@ypapouin I have added a deployment diagram and some getting started instructions in the README. I hope this helps.

If you want a single node cluster that you manage yourself, microk8s is an easy solution (together with k9s to explore the cluster content).

I see you mention GitLab. Note that currently runboat supports GitHub only. It should not be too difficult to support GitLab, but some light refactoring is necessary to make it support multiple git providers.

**Comment by ypapouin** [comment]
Thank you @sbidoul.

Indeed microk8s is a lot simpler and I managed to run it and start a Runboat instance on docker.
The diagram make the whole architecture more clear.

I took a look on the runboat source code and I agree that making support for GitLab would not be very hard.
But before starting hacking on it, I would like to have a fully working infrastructure via Github.

Could you add some screenshots of the current GitHub Webhook settings for an sample repository (url, authentication, etc.) ?

**Comment by sbidoul** [comment]
Here is the GitHub webhook config for OCA:

- Payload URL: https://runboat.odoo-community.org/webhooks/github
- Secret: the same as `RUNBOAT_GITHUB_WEBHOOK_SECRET`
- Events: Pushes and Pull Requests
